docs/content/tutorials/custom-listen-ports.md

```diff
@@ -0,0 +1,137 @@
+## Customizing the `listen` line in NGINX Ingress Controller.
+
+This document will explain how to change the default ports that NGINX Ingress Controller is configured for, as well as add additional `listen` settings. For more information, please read the [NGINX Listen documentation](http://nginx.org/en/docs/http/ngx_http_core_module.html#listen).
+
+
+## Changing Default Ports
+
+By default, NGINX Ingress Controller listens on ports 80 and 443. These ports can be changed easily, but modifying the `listen` ports for your NGINX Ingress resources will require the editing of .tmpl files.
+
+If you are using `ingress` resource you will need to modify:
+- `nginx-plus-ingress.tmpl` if using NGINX Plus
+- `nginx-ingress.tmpl` if using NGINX OSS
+
+If you are using NGINX Ingress Controller CRDs (virtualServer):
+- `nginx-plus-virtualserver.tmpl` for NGINX Plus
+- `nginx-virtualserver.tmpl` if using NGINX OSS
+
+For this example, we are going to use the `nginx-virtualserver.tmpl` to change the port from 80 to 85.
+
+Here we modify `nginx-virtualserver.tmpl` to change the port setting:
+
+```
+server {
+    listen 80{{ if $s.ProxyProtocol }} proxy_protocol{{ end }};
+
+    server_name {{ $s.ServerName }};
+
+    set $resource_type "virtualserver";
+    set $resource_name "{{$s.VSName}}";
+    set $resource_namespace "{{$s.VSNamespace}}";
+```
+To change the listen port from `80` to `85`, we modify the `listen` line at the start of the server configuration block.
+
+It would then look like this:
+```
+server {
+    listen 85{{ if $s.ProxyProtocol }} proxy_protocol{{ end }};
+
+    server_name {{ $s.ServerName }};
+
+    set $resource_type "virtualserver";
+    set $resource_name "{{$s.VSName}}";
+    set $resource_namespace "{{$s.VSNamespace}}";
+```
+
+Edit the file you need (per the example above). In my case, I edited, `nginx-plus-virtualserver.tmpls`:
+
+
+## Rebuild your NGINX Ingress controller image
+
+You will need to build your new NGINX Ingress controller image for the new port settings to take affect.
+Once the image is built and pushed, make sure you update your deployment to point to the new image and deploy.
+Once deployed, create a new `virtualServer` resource and then run `nginx -T` to see if the port takes affect.
```

Comment:
maybe add full command:
`kubectl exec -it -n nginx-ingress <pod_name> -- nginx -T`